river/Cursor.zig

```diff
@@ -1095,3 +1099,44 @@ fn passthrough(self: *Self, time: u32) void {
         self.clearFocus();
     }
 }
+
+pub fn warp(self: *Self) void {
+    self.needWarp = false;
+    if (self.seat.focused_output == &server.root.noop_output) return;
+    // Warp pointer to center of the focused view/output (In layout coordinates) if enabled.
+    var layout_box: wlr.Box = undefined;
+    server.root.output_layout.getBox(self.seat.focused_output.wlr_output, &layout_box);
+    var box = switch (server.config.warp_cursor) {
+        .disabled => return,
+        .@"on-output-change" => layout_box,
+        .@"on-focus-change" => switch (self.seat.focused) {
+            .layer, .lock_surface, .none => layout_box,
+            .view => |view| wlr.Box{
+                .x = layout_box.x + view.current.box.x,
+                .y = layout_box.y + view.current.box.y,
+                .width = @intCast(c_int, view.current.box.width),
+                .height = @intCast(c_int, view.current.box.height),
+            },
+            .xwayland_override_redirect => |or_window| wlr.Box{
+                .x = or_window.xwayland_surface.x,
+                .y = or_window.xwayland_surface.y,
+                .width = or_window.xwayland_surface.width,
+                .height = or_window.xwayland_surface.height,
+            },
+        },
+    };
+    const usable_box = self.seat.focused_output.usable_box;
+    const usable_layout_box = wlr.Box{
+        .x = layout_box.x + usable_box.x,
+        .y = layout_box.y + usable_box.y,
+        .width = usable_box.width,
+        .height = usable_box.height,
+    };
+    if (!layout_box.containsPoint(self.wlr_cursor.x, self.wlr_cursor.y) or (usable_layout_box.containsPoint(self.wlr_cursor.x, self.wlr_cursor.y) and !box.containsPoint(self.wlr_cursor.x, self.wlr_cursor.y))) {
```

Comment:
This should be split across multiple lines for readability, `zig fmt` allows line breaks after binary operators such as `and`/`or`.

Also, what is the purpose of the useable_layout_box check here?

Reply:
it prevents the mouse to be warpped if it was outside the usable area, this prevents the mouse being yanked when say you click the bar to select tag